cmd/cleanup/cleanup.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/sirupsen/logrus"
+)
+
+func main() {
+	rp, ok := os.LookupEnv("REDIS_PROVIDER")
+	if !ok {
+		logrus.Fatal("missing connection string to redis")
+		return
+	}
+	opt, err := redis.ParseURL(rp)
+	if err != nil {
+		logrus.WithError(err).Fatal("Could not connect to Redis, exists")
+		return
+	}
+	redisClient := redis.NewClient(opt)
+
+	_, err = redisClient.Ping(context.Background()).Result()
+	if err != nil {
+		logrus.WithError(err).Fatal("Could talk to Redis, exists")
+		return
+	}
+
+	if len(os.Args) > 2 {
+		group := os.Args[1]
+		stream := os.Args[2]
+		// prune old inactive consumers
+		cu := redisClient.XInfoConsumers(context.Background(), stream, group)
+		cs, err := cu.Result()
+		if err != nil {
+			logrus.WithError(err).Fatalf("failed to list consumers for stream: %q group: %q", stream, group)
+		}
+		fmt.Printf("Found %d consumers \n", len(cs))
+		for x, consumer := range cs {
+			fmt.Printf("%d: Checking consumer: %q : ", x, consumer.Name)
+			if consumer.Pending == 0 && // don't remove with pending msg
+				consumer.Idle > time.Hour.Milliseconds()*24 { // respect PruneTimeout before trying to delete
```

Comment:
Is this related to `PruneTimeout` in `consumer.go`? If not, why 24 milliseconds?

Reply:
i should change the comment, but the time i 1 hour in millis * 24 = 1 day worth of idle

Reply:
I read it too fast, missed the hour :D